src/org/thoughtcrime/securesms/WebViewActivity.java

```diff
@@ -122,6 +127,22 @@ public WebResourceResponse shouldInterceptRequest(WebView view, WebResourceReque
         return super.shouldInterceptRequest(view, request);
       }
     });
+    webView.setWebChromeClient(new WebChromeClient() {
+      @Override
+      @RequiresApi(21)
+      public boolean onShowFileChooser(WebView webView, ValueCallback<Uri[]> filePathCallback, WebChromeClient.FileChooserParams fileChooserParams) {
+        if (mFileUploadCallbackSecond != null) {
+          mFileUploadCallbackSecond.onReceiveValue(null);
+        }
+        mFileUploadCallbackSecond = filePathCallback;
+        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+        intent.setType("*/*");
+        intent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, fileChooserParams.getMode() == FileChooserParams.MODE_OPEN_MULTIPLE);
+        startActivityForResult(Intent.createChooser(intent, getString(R.string.select)), REQUEST_CODE_FILE_PICKER);
```

Comment:
does compiler complain about that? why is it needed?

Reply:
no but it is clearer for people using a dummy plain text editor like me :D to avoid ambiguity and clearly see from where it is that method from, better explicit than implicit

#AndroidStudioKillsMyMachine

Reply:
i see :)

as it is needed for having `filePathCallback` as membername as well as parameter anyways (see next commit), this even improves consistency.